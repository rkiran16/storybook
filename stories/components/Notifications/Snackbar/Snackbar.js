import { html } from "lit-html";
import { IconButton } from "../../IconButton/IconButton";

const Icon = (type) => {
  switch (type) {
    case "warning":
      return html` <span class="material-icons"> warning_amber </span> `;
    case "success":
      return html` <span class="material-icons"> check_circle_outline </span> `;
    default:
      return html` <span class="material-icons"> error_outline </span> `;
  }
};

export const Snackbar = ({
  title,
  label,
  href,
  linkTitle,
  color = "primary",
  icon = "default",
  button = false,
}) => {
  return html`
    <div class="alert alert-custom  bg-${color} alert-dismissible pe-3 text-white"role="alert">
      <div class="d-flex flex-row justify-content-between align-items-start align-items-md-center w-100 ">
         <div class="d-flex align-items-start align-items-lg-center me-2">${icon ? Icon(icon) : null}</div>
          <div class="d-flex flex-column flex-sm-row align-items-sm-center w-100">
            <span>
                <strong>${title}</strong>
                <span class="lh-1">${label}</span>
            </span>
            <div class="d-flex justify-content-between align-items-center w-100 mt-2 mt-sm-0">
              ${linkTitle ? html`<a href="#" class="text-white ms-sm-2 inline">${linkTitle}</a>` : ''}
              ${button ? html `<button type="button" class="btn btn-icon btn-icon-w-text horizontal inverse">
              Undo
              <span class="material-icons me-1">replay</span>
            </button>` : ''}
            </div>
        </div>
        <div class="d-flex">
          <button type="button" class="btn btn-icon w-auto h-auto text-white" data-bs-dismiss="alert" aria-label="Close">
            <span class="material-icons">close</span>
          </button>
        </div>
      </div>
    </div>
  `;
}
