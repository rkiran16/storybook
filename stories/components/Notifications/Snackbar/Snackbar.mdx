import { Canvas, Meta, Story } from "@storybook/addon-docs";

# Snackbar

Snackbars are non-modal, time-based window elements used to display short messages. They are more disruptive than inline notifications and are best used with system-generated messages that do not correspond to a specific section of the UI. Their color and icon should correspond with the message intent.

Message

- Keep messages clear and concise (not to exceed 3 lines)

Sizing

- Fixed width 8 columns centered on desktop (1064px) and should not be expanded to fit the content area
- Height expands to fit second line of copy

Placement

- Snackbars should be visible no matter where a user is on the screen. Snackbars are center-aligned 24px from the top edge of the screen.
- They do cover content
- If multiple snackbars are needed, snackbars stack with 8px between them and the most recent appears on top.

Dismissal

- Snackbars timeout automatically after 20 seconds on the screen with the ability for the development to change the timing within their page level work
  -They can include a close button so users can dismiss them sooner

Common uses

- confirmation that an action has taken place
  - new cart created
  - new list created
  - deletion of an item
- action not able to be completed (user tries to complete an action and something on the backend prevents it from happening)

How to handle batch actions
If one action effects multiple items (ex: deleting multiple lists), use one success message for all.

Snackbars with actions
Sometimes an action needs to be included in a snacker. This action should not be imperative to the user as the snack bar times out after 20 seconds with the ability for the development to change the timing within their page level work . Links and tertiary buttons should be no more than three words.

<Canvas>
  <Story id="components-notifications-snackbar--default"></Story>
  <Story id="components-notifications-snackbar--with-action"></Story>
  <Story id="components-notifications-snackbar--error"></Story>
  <Story id="components-notifications-snackbar--warning"></Story>
  <Story id="components-notifications-snackbar--info"></Story>
  <Story id="components-notifications-snackbar--with-button"></Story>
  <Story id="components-notifications-snackbar--with-button-and-action"></Story>
</Canvas>
