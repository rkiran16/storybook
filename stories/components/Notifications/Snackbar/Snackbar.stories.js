import { Snackbar } from "./Snackbar";
import { html } from "lit-html";
import docs from "./Snackbar.mdx";

export default {
  title: "Components/Notifications/Snackbar",
  parameters: {
    zeplinLink:
      "  https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d1c10223a41149c68958",
    docs: {
      page: docs,
    },
  },
  argTypes: {
    color: {
      name: "Color",
      control: { type: "select" },
      defaultValue: "success",
      options: ["success", "danger", "warning", "info"],
    },
    icon: {
      name: "Icon",
      control: { type: "select" },
      defaultValue: "default",
      options: ["none", "default", "warning", "success"],
      mapping: {
        none: false,
        default: "same",
        warning: "warning",
        success: "success",
      },
    },
  },
};

const Template = (args) => html`${Snackbar(args)}`;

export const Default = Template.bind({});
Default.args = {
  title: "Success!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "success",
  icon: "success",
};

export const WithAction = Template.bind({});
WithAction.args = {
  title: "Success!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "success",
  icon: "success",
  href: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d1c10223a41149c68958",
  linkTitle: "Link Action",
};
export const Error = Template.bind({});
Error.args = {
  title: "Error!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "danger",
};
export const Warning = Template.bind({});
Warning.args = {
  title: "Warning!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "warning",
  icon: "warning",
};
export const Info = Template.bind({});
Info.args = {
  title: "Info!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "info",
};

export const WithButton = Template.bind({});
WithButton.args = {
  title: "Info!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "info",
  button: true,
};

export const WithButtonAndAction = Template.bind({});
WithButtonAndAction.args = {
  title: "Info!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "info",
  button: true,
  href: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d1c10223a41149c68958",
  linkTitle: "Link Action",
};
