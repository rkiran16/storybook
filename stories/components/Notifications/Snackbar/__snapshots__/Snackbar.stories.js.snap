// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Notifications/Snackbar Default 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <div class=\\"alert alert-custom  bg-",
        " alert-dismissible pe-3 text-white\\"role=\\"alert\\">
      <div class=\\"d-flex flex-row justify-content-between align-items-start align-items-md-center w-100 \\">
         <div class=\\"d-flex align-items-start align-items-lg-center me-2\\">",
        "</div>
          <div class=\\"d-flex flex-column flex-sm-row align-items-sm-center w-100\\">
            <span>
                <strong>",
        "</strong>
                <span class=\\"lh-1\\">",
        "</span>
            </span>
            <div class=\\"d-flex justify-content-between align-items-center w-100 mt-2 mt-sm-0\\">
              ",
        "
              ",
        "
            </div>
        </div>
        <div class=\\"d-flex\\">
          <button type=\\"button\\" class=\\"btn btn-icon w-auto h-auto text-white\\" data-bs-dismiss=\\"alert\\" aria-label=\\"Close\\">
            <span class=\\"material-icons\\">close</span>
          </button>
        </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "success",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> check_circle_outline </span> ",
          ],
          "values": Array [],
        },
        "Success!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        "",
        "",
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Snackbar Error 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <div class=\\"alert alert-custom  bg-",
        " alert-dismissible pe-3 text-white\\"role=\\"alert\\">
      <div class=\\"d-flex flex-row justify-content-between align-items-start align-items-md-center w-100 \\">
         <div class=\\"d-flex align-items-start align-items-lg-center me-2\\">",
        "</div>
          <div class=\\"d-flex flex-column flex-sm-row align-items-sm-center w-100\\">
            <span>
                <strong>",
        "</strong>
                <span class=\\"lh-1\\">",
        "</span>
            </span>
            <div class=\\"d-flex justify-content-between align-items-center w-100 mt-2 mt-sm-0\\">
              ",
        "
              ",
        "
            </div>
        </div>
        <div class=\\"d-flex\\">
          <button type=\\"button\\" class=\\"btn btn-icon w-auto h-auto text-white\\" data-bs-dismiss=\\"alert\\" aria-label=\\"Close\\">
            <span class=\\"material-icons\\">close</span>
          </button>
        </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "danger",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "Error!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        "",
        "",
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Snackbar Info 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <div class=\\"alert alert-custom  bg-",
        " alert-dismissible pe-3 text-white\\"role=\\"alert\\">
      <div class=\\"d-flex flex-row justify-content-between align-items-start align-items-md-center w-100 \\">
         <div class=\\"d-flex align-items-start align-items-lg-center me-2\\">",
        "</div>
          <div class=\\"d-flex flex-column flex-sm-row align-items-sm-center w-100\\">
            <span>
                <strong>",
        "</strong>
                <span class=\\"lh-1\\">",
        "</span>
            </span>
            <div class=\\"d-flex justify-content-between align-items-center w-100 mt-2 mt-sm-0\\">
              ",
        "
              ",
        "
            </div>
        </div>
        <div class=\\"d-flex\\">
          <button type=\\"button\\" class=\\"btn btn-icon w-auto h-auto text-white\\" data-bs-dismiss=\\"alert\\" aria-label=\\"Close\\">
            <span class=\\"material-icons\\">close</span>
          </button>
        </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "info",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "Info!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        "",
        "",
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Snackbar Warning 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <div class=\\"alert alert-custom  bg-",
        " alert-dismissible pe-3 text-white\\"role=\\"alert\\">
      <div class=\\"d-flex flex-row justify-content-between align-items-start align-items-md-center w-100 \\">
         <div class=\\"d-flex align-items-start align-items-lg-center me-2\\">",
        "</div>
          <div class=\\"d-flex flex-column flex-sm-row align-items-sm-center w-100\\">
            <span>
                <strong>",
        "</strong>
                <span class=\\"lh-1\\">",
        "</span>
            </span>
            <div class=\\"d-flex justify-content-between align-items-center w-100 mt-2 mt-sm-0\\">
              ",
        "
              ",
        "
            </div>
        </div>
        <div class=\\"d-flex\\">
          <button type=\\"button\\" class=\\"btn btn-icon w-auto h-auto text-white\\" data-bs-dismiss=\\"alert\\" aria-label=\\"Close\\">
            <span class=\\"material-icons\\">close</span>
          </button>
        </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "warning",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> warning_amber </span> ",
          ],
          "values": Array [],
        },
        "Warning!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        "",
        "",
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Snackbar With Action 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <div class=\\"alert alert-custom  bg-",
        " alert-dismissible pe-3 text-white\\"role=\\"alert\\">
      <div class=\\"d-flex flex-row justify-content-between align-items-start align-items-md-center w-100 \\">
         <div class=\\"d-flex align-items-start align-items-lg-center me-2\\">",
        "</div>
          <div class=\\"d-flex flex-column flex-sm-row align-items-sm-center w-100\\">
            <span>
                <strong>",
        "</strong>
                <span class=\\"lh-1\\">",
        "</span>
            </span>
            <div class=\\"d-flex justify-content-between align-items-center w-100 mt-2 mt-sm-0\\">
              ",
        "
              ",
        "
            </div>
        </div>
        <div class=\\"d-flex\\">
          <button type=\\"button\\" class=\\"btn btn-icon w-auto h-auto text-white\\" data-bs-dismiss=\\"alert\\" aria-label=\\"Close\\">
            <span class=\\"material-icons\\">close</span>
          </button>
        </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "success",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> check_circle_outline </span> ",
          ],
          "values": Array [],
        },
        "Success!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<a href=\\"#\\" class=\\"text-white ms-sm-2 inline\\">",
            "</a>",
          ],
          "values": Array [
            "Link Action",
          ],
        },
        "",
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Snackbar With Button 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <div class=\\"alert alert-custom  bg-",
        " alert-dismissible pe-3 text-white\\"role=\\"alert\\">
      <div class=\\"d-flex flex-row justify-content-between align-items-start align-items-md-center w-100 \\">
         <div class=\\"d-flex align-items-start align-items-lg-center me-2\\">",
        "</div>
          <div class=\\"d-flex flex-column flex-sm-row align-items-sm-center w-100\\">
            <span>
                <strong>",
        "</strong>
                <span class=\\"lh-1\\">",
        "</span>
            </span>
            <div class=\\"d-flex justify-content-between align-items-center w-100 mt-2 mt-sm-0\\">
              ",
        "
              ",
        "
            </div>
        </div>
        <div class=\\"d-flex\\">
          <button type=\\"button\\" class=\\"btn btn-icon w-auto h-auto text-white\\" data-bs-dismiss=\\"alert\\" aria-label=\\"Close\\">
            <span class=\\"material-icons\\">close</span>
          </button>
        </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "info",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "Info!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        "",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<button type=\\"button\\" class=\\"btn btn-icon btn-icon-w-text horizontal inverse\\">
              Undo
              <span class=\\"material-icons me-1\\">replay</span>
            </button>",
          ],
          "values": Array [],
        },
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Snackbar With Button And Action 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <div class=\\"alert alert-custom  bg-",
        " alert-dismissible pe-3 text-white\\"role=\\"alert\\">
      <div class=\\"d-flex flex-row justify-content-between align-items-start align-items-md-center w-100 \\">
         <div class=\\"d-flex align-items-start align-items-lg-center me-2\\">",
        "</div>
          <div class=\\"d-flex flex-column flex-sm-row align-items-sm-center w-100\\">
            <span>
                <strong>",
        "</strong>
                <span class=\\"lh-1\\">",
        "</span>
            </span>
            <div class=\\"d-flex justify-content-between align-items-center w-100 mt-2 mt-sm-0\\">
              ",
        "
              ",
        "
            </div>
        </div>
        <div class=\\"d-flex\\">
          <button type=\\"button\\" class=\\"btn btn-icon w-auto h-auto text-white\\" data-bs-dismiss=\\"alert\\" aria-label=\\"Close\\">
            <span class=\\"material-icons\\">close</span>
          </button>
        </div>
      </div>
    </div>
  ",
      ],
      "values": Array [
        "info",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons\\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "Info!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, Lorem ipsum dolor sit amet, consectetur adipiscing elit,Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<a href=\\"#\\" class=\\"text-white ms-sm-2 inline\\">",
            "</a>",
          ],
          "values": Array [
            "Link Action",
          ],
        },
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<button type=\\"button\\" class=\\"btn btn-icon btn-icon-w-text horizontal inverse\\">
              Undo
              <span class=\\"material-icons me-1\\">replay</span>
            </button>",
          ],
          "values": Array [],
        },
      ],
    },
  ],
}
`;
